lib/active_record_host_pool/connection_handling.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module ActiveRecordHostPool
+  # ActiveRecord 6.0 introduces multiple database support. This introduces a
```

Comment:
I had to re-read this description a bunch of times before I really understood what was happening. This may have just been me though, so feel free to ignore my suggestions below 😅 


> ActiveRecord 6.0 introduced multiple database support. With that, an update has been made in rails/rails#35089
to ensure that all query caches are cleared across connection handlers and pools. If you write on one connection, the other connection will have the update that occurred.

> This broke ARHP which implements its own pool, allowing you to access multiple databases with the same connection (e.g. 1 connection for 100 shards on the same server).

> This patch maintains the reference to the database during the cache clearing to ensure that the database doesn't get swapped out mid-way into an operation.

> This is a private Rails API and may change in future releases as they're actively working on sharding in Rails 6 and above.

Reply:
@mriddle, I really like what you suggested above. That definitely seems much clearer. Updated in f840ccc0c97bf84eb71b17aff96f478764fa308d